packages/langium-sql/src/sql.langium

```diff
@@ -272,6 +273,9 @@ OverClause:
     'OVER' ('(' spec=WindowSpec ')' | windowName=[WindowSpec:Identifier])
 ;
 
+WhenClause:
+    'CASE' 'WHEN' condition=Expression 'THEN' result=Expression
```

Comment:
CASE should be outside the repetition: https://www.w3schools.com/sql/sql_case.asp
Each CASE has a corresponding END:

```sql
CASE
    WHEN condition1 THEN result1
    WHEN condition2 THEN result2
    WHEN conditionN THEN resultN
    ELSE result
END;
```